sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -343,6 +368,25 @@ class JacksonParser(
       record: T,
       createParser: (JsonFactory, T) => JsonParser,
       recordLiteral: T => UTF8String): Seq[InternalRow] = {
+    parseWithArrayOfPrimitiveSupport(record, createParser, recordLiteral) match {
+      case rows: Seq[InternalRow] => rows
+      case _: Seq[_] => throw BadRecordException(() => recordLiteral(record), () => None,
+        new RuntimeException("Conversion of array of primitive data is not yet supported here."))
+    }
+  }
+
+  /**
+   * Parse the JSON input. This function can return a set of [[InternalRow]]s
+   * if a [[StructType]] is defined as schema, otherwise it returns a set of
+   * objects.
```

Comment:
Please add comment that this is used when passing `ArrayType` of primitive types.

Reply:
Btw, and also add comment to existing `parse` to clarify its usage. It might not easily to know which to call at the first glance.